### PR TITLE
[jax2tf] First draft for the conversion of FFTs.

### PR DESCRIPTION
**Pull request by bchetioui**
Translation does not seem to cause any problem, however errors in
calls to lax.lax_fft are handled at a higher abstraction layer
(in jax.numpy.fft). Maybe it would thus make sense to make the
tests call jax.numpy.fft._fft_core instead of lax.lax_fft to ensure
the inputs are sanitized in the same way?

### CODE REVIEW

File: jax/experimental/jax2tf/jax2tf.py [hunk 1]
```diff
@@ -1167,6 +1169,16 @@ def _sort(*operand: TfVal, dimension: int, is_stable: bool, num_keys: int) -> Tu
 
 tf_impl[lax.sort_p] = _sort
 
+def _fft(x, fft_type, fft_lengths):
+  tf_funcs = {FftType.FFT: [tf.signal.fft, tf.signal.fft2d, tf.signal.fft3d],
+              FftType.IFFT: [tf.signal.ifft, tf.signal.ifft2d, tf.signal.ifft3d],
+              FftType.RFFT: [tf.signal.rfft, tf.signal.rfft2d, tf.signal.rfft3d],
+              FftType.IRFFT: [tf.signal.irfft, tf.signal.irfft2d, tf.signal.irfft3d]}
+
+  return tf_funcs[fft_type][len(fft_lengths) - 1](x)
```
Comment (gnecula) [inline jax/experimental/jax2tf/jax2tf.py:1178]: Let's discuss this offline: the only thing we need from `fft_lengths` is its length?

File: jax/experimental/jax2tf/tests/primitive_harness.py [hunk 2]
```diff
@@ -405,6 +407,120 @@ def parameterized(harness_group: Iterable[Harness],
...
+          shape=shape,
+          dtype=dtype,
+          fft_type=fft_type,
+          fft_lengths=fft_lengths)
+  for dtype in jtu.dtypes.complex
+  for shape in [(14, 15, 16)]
+  for fft_type, fft_lengths in [(FftType.FFT, shape[-3:]),
+                                (FftType.IFFT, shape[-3:]),
+                                (FftType.IRFFT, shape[-3:-1] + ((shape[-1] - 1) * 2,))]
+) + tuple( # 4D test, no complex, should error
```
Comment (gnecula) [inline jax/experimental/jax2tf/tests/primitive_harness.py:495]: Perhaps we can merge the non-complex and complex 4D tests?

File: jax/experimental/jax2tf/tests/primitive_harness.py [hunk 3]
```diff
@@ -405,6 +407,120 @@ def parameterized(harness_group: Iterable[Harness],
...
+  Harness(f"4d_shape={jtu.format_shape_dtype_string(shape, dtype)}_ffttype={fft_type}_fftlengths={fft_lengths}",
+          lax.lax_fft.fft,
+          [RandArg(shape, dtype), StaticArg(fft_type), StaticArg(fft_lengths)],
+          shape=shape,
+          dtype=dtype,
+          fft_type=fft_type,
+          fft_lengths=fft_lengths)
+  for dtype in [t for t in jtu.dtypes.all if t not in jtu.dtypes.complex]
+  for shape in [(14, 15, 16, 17)]
+  for fft_type, fft_lengths in [(FftType.FFT, shape[-4:]),
```
Comment (gnecula) [inline jax/experimental/jax2tf/tests/primitive_harness.py:505]: Here `shape[-4:]` is the same as `shape`, right?

Comment (SIben) [reply jax/experimental/jax2tf/tests/primitive_harness.py]: Yes! Addressed.

File: jax/experimental/jax2tf/tests/primitive_harness.py (hunk 2 above)
Comment (SIben) [reply jax/experimental/jax2tf/tests/primitive_harness.py]: Merged all the complex/non-complex tests together.

File: jax/experimental/jax2tf/jax2tf.py (hunk 1 above)
Comment (gnecula) [reply jax/experimental/jax2tf/jax2tf.py:1178]: We had discussed online that we would add an assertion that `fft_lengths` has the expected values.

**Pull request merged**